include/http/router.js

```diff
@@ -1,197 +1,276 @@
+    //pb dependencies
+    var RequestHandler = pb.RequestHandler;
 
-            this.router = new Router();
+    /**
+     * Responsible for routing a request through the registered middleware to serve up a response
+     * @class Router
+     * @constructor
+     * @param {Request} req
+     * @param {Response} res
+     */
+    class Router {
+        constructor(req, res) {
 
-            this.app.use(bodyParser({
-                multipart: true,
-                // formidable: { uploadDir: path.join(__dirname, 'tmp') }
-            }));
-            let passport = Passport(pb);
-            this.app.use(Session(this.app));
-            this.app.use(Cookies());
-            this.app.use(passport.initialize());
-            this.app.use(passport.session());
+            /**
+             * Represents the current position of the middleware that is currently executing
+             * @property index
+             * @type {number}
+             */
+            this.index = 0;
+
+            /**
+             * @property req
+             * @type {Request}
+             */
+            this.req = req;
+
+            /**
+             * @property res
+             * @type {Response}
+             */
+            this.res = res;
         }
 
-        static registerRoute(routeDescriptor) {
-            this.internalRouteList.push(routeDescriptor);
+        /**
+         * Starts the execution of the middleware pipeline against the specified request/response pair
+         * @method handle
+         */
+        handle() {
+
+            //set reference to the handler
+            this.req.handler = new RequestHandler(this.req, this.res);
+            this.req.router = this;
+
+            return this._handle(this.req, this.res).catch((err) => {
+                this.res.statusCode = 500;
+                this.res.end();
+            });
         }
 
-        static addMiddlewareAfter(name, middleware) {
-            let index = this._indexOfMiddleware(name);
-            if (index >= 0) {
-                return this._addMiddlewareAt(index + 1, middleware);
+        _handleMiddleware(req, res, action) {
+            return new Promise((resolve, reject) => {
+                process.nextTick(() => {
+                    try {
+                        action(req, res, (err) => {
+                            if (err) reject(err);
+                            else resolve();
+                        });
+                    } catch (e) {
+                        reject(e);
+                    }
+                });
+            });
+        }
+
+        async _handleMiddlewares (req, res) {
+            try {
+                for (; this.index < Router.middleware.length; this.index++) {
+                    await this._handleMiddleware(req, res, Router.middleware[this.index].action);
+                }
+            } catch (e) {
+                return new Promise((resolve, reject) => {
```

Comment:
adding a log would not be a bad idea since we are in a critical failure state at this point.  wouldnt be bad to look and see.  Btw the worst offending domain is the one you copied that code from @josesubi Hopefully that has been removed.